lib/gis_robot_suite/structural_file_builder.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+module GisRobotSuite
+  # Builds cocina file metadata
+  class StructuralFileBuilder
+    def self.build(*args)
+      new(*args).build
+    end
+
+    # @param [Hash<Symbol,Assembly::ObjectFile>] objects
+    def initialize(assembly_objectfile, access, version)
+      @assembly_objectfile = assembly_objectfile
+      @access = access
+      @version = version
+    end
+
+    attr_reader :assembly_objectfile, :access, :version
+
+    def build
+      Cocina::Models::File.new(file_attributes)
+    end
+
+    private
+
+    def file_attributes
+      {
+        hasMimeType: mimetype,
+        administrative: { publish: true, shelve: true, sdrPreserve: roletype_for_file == 'master' },
+        access: access,
+        externalIdentifier: "https://cocina.sul.stanford.edu/file/#{SecureRandom.uuid}",
+        type: Cocina::Models::ObjectType.file,
+        version: version,
+        filename: assembly_objectfile.filename,
+        label: assembly_objectfile.filename,
+        size: assembly_objectfile.filesize,
+        use: roletype_for_file || 'master',
+        hasMessageDigests: [
+          { type: 'sha1', digest: assembly_objectfile.sha1 },
+          { type: 'md5', digest: assembly_objectfile.md5 }
+        ]
+      }
+    end
+
+    def mimetype
+      @mimetype ||= assembly_objectfile.mimetype
+    end
+
+    def roletype_for_file
+      @roletype_for_file ||= if mimetype == 'application/zip'
+                               if assembly_objectfile.path =~ /_(EPSG_\d+)/i # derivative
+                                 'derivative'
+                               else
+                                 'master'
+                               end
+                             end || nil
```

Comment:
"else nil" would be a bit clearer than "|| nil".

Reply:
agreed, happy to fix.  was a holdover from the original PR (and i think the original refactored code).  but i also found it a bit hard to read.

Reply:
ah, right, as i'd initially suspected, and as rubocop suggested when i tried this change, the `|| nil` / `else nil` is redundant anyway, because if the `if` condition is false-y, the `if` statement returns `nil` anyway, e.g. 

```ruby
irb(main):001* foo = if 1 == 2
irb(main):002*   true
irb(main):003> end
=> nil
irb(main):004> foo
=> nil
irb(main):005> quit
```

so i just dropped the `|| nil` entirely.